app/controllers/application_controller.rb

```diff
@@ -188,9 +188,35 @@ def service_provider_mfa_setup_url
     service_provider_mfa_policy.user_needs_sp_auth_method_setup? ? two_factor_options_url : nil
   end
 
+  def fix_broken_personal_key_url
+    if current_user.broken_personal_key?
```

Comment:
Maybe a style preference, but how would you feel about an early return here, to avoid indenting the whole method?

```suggestion
    return if !current_user.broken_personal_key?
```